### PR TITLE
PLANET-6960 Add sidebar option to define canonical link

### PR DESCRIPTION
**Pull request by Osong-Michael**
**Description**

See [PLANET-6960](https://jira.greenpeace.org/browse/PLANET-6960)

Associated [PR](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/986)

### Testing
On your local or [Umbriel Test Instance](https://www-dev.greenpeace.org/test-umbriel/), you can add the Canonical link from the sidebar (_Search Engine Optimizations_ section), to any post type and update the page. On reload in the editor you should still see the data that you entered, and when inspecting the page in the frontend you should see the correct link tag added to the head.

Screenshot 2022-12-14 at 03 36 45

Screenshot 2022-12-14 at 03 41 30

### CODE REVIEW

File: src/Post.php [hunk 1]
```diff
@@ -350,6 +350,7 @@ public function get_og_image() {
 		return [];
 	}
 
+
```
Comment (mleray) [inline src/Post.php:353]: Not a big deal, but this extra white space is not needed 😉

File: functions.php [hunk 2]
```diff
@@ -474,3 +474,16 @@ function() {
 		}
 	}
 );
+
+/**
+ * Adds the canonical link from the sidebar menu to the post head
+ */
+function add_meta_tags() {
+	global $post;
+	if ( isset( $post->p4_seo_canonical_url ) ) {
```
Comment (mleray) [inline functions.php:483]: This seems to return true also if `p4_seo_canonical_url` is an empty string (so for example if it was set in the backend at some point and then removed), in that case you end up with a link with an empty href

Comment (Osong-Michael) [reply functions.php]: Thank you, just resolved this.

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [venus](https://www-dev.greenpeace.org/test-venus) | [admin](https://www-dev.greenpeace.org/test-venus/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-venus/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-venus) | [composer-local.json](https://github.com/greenpeace/planet4-test-venus/blob/main/composer-local.json)

:watch: 2022.12.15 08:21:09

File: src/Post.php (hunk 1 above)
Comment (Osong-Michael) [reply src/Post.php]: 😄 Done.

**Review by mleray — APPROVED** [review]
Looks good to me 😄 👍

**Pull request merged**